### PR TITLE
[FEATURE] fsInterface: Add mkdir function

### PR DESCRIPTION
**Pull request by RandomByte**

### CODE REVIEW

File: lib/AbstractReader.js [hunk 1]
```diff
@@ -21,17 +21,17 @@ class AbstractReader {
 	}
 
 	/**
-	 * Locates resourcess by GLOB.
+	 * Locates resources by matching GLOB patterns.
```
Comment (matz3) [inline lib/AbstractReader.js]: I would also use "glob" not "GLOB". At least that's how it's spelled everywhere: https://en.wikipedia.org/wiki/Glob_(programming)

Comment (RandomByte) [reply lib/AbstractReader.js]: Done

**Comment by RandomByte** [comment]
Added one more small and not really related change 😅

File: lib/tracing/traceSummary.js [hunk 2]
```diff
@@ -60,6 +63,9 @@ function someTraceStarted() {
 
 function someTraceEnded() {
 	return new Promise(function(resolve, reject) {
+		if (!active) {
+			resolve();
```
Comment (matz3) [inline lib/tracing/traceSummary.js]: Missing return

Comment (RandomByte) [reply lib/tracing/traceSummary.js]: Done

Comment (RandomByte) [reply lib/tracing/traceSummary.js]: And added tests

**Comment by RandomByte** [comment]
Added `mock-require` for usage in the traceSummary tests

**Comment by RandomByte** [comment]
Required for https://github.com/SAP/ui5-builder/pull/42

**Pull request merged**